op-supervisor/supervisor/backend/cross/cycle_test.go

```diff
@@ -354,52 +356,41 @@ func TestHazardCycleChecksNoCycle(t *testing.T) {
 			msg: "expected no cycle found first log is exec",
 		},
 		{
-			name: "cycle through older timestamp",
+			name: "no cycle using different timestamp",
 			chainBlocks: map[string]chainBlockDef{
 				"1": {
 					logCount: 2,
 					messages: map[uint32]*types.ExecutingMessage{
-						0: execMsg("2", 0),
-						1: execMsgWithTimestamp("2", 1, 101),
-					},
-				},
-				"2": {
-					logCount: 2,
-					messages: map[uint32]*types.ExecutingMessage{
-						0: execMsg("1", 1),
+						0: execMsgWithTimestamp("1", 1, testDefaultTimestamp+1),
 					},
 				},
 			},
-			msg: "expected no cycle detection error for cycle through messages with different timestamps",
+			msg: "expected no cycle for different timestamp",
 		},
-		// This should be caught by earlier validations, but included for completeness.
+	}
+	runHazardCycleChecksTestCaseGroup(t, "NoCycle", tests)
+}
+
+func TestHazardCycleChecksCycle(t *testing.T) {
+	tests := []hazardCycleChecksTestCase{
 		{
-			name: "cycle through younger timestamp",
+			// 0->1->2->0
+			name: "3-cycle in single chain",
 			chainBlocks: map[string]chainBlockDef{
 				"1": {
-					logCount: 2,
-					messages: map[uint32]*types.ExecutingMessage{
-						0: execMsg("2", 0),
-						1: execMsgWithTimestamp("2", 1, 99),
-					},
-				},
-				"2": {
-					logCount: 2,
+					logCount: 3,
 					messages: map[uint32]*types.ExecutingMessage{
-						0: execMsg("1", 1),
+						0: execMsg("1", 2),
 					},
 				},
 			},
-			msg: "expected no cycle detection error for cycle through messages with different timestamps",
+			expectErr: ErrCycle,
+			msg:       "expected cycle detection error",
 		},
-	}
-	runHazardCycleChecksTestCaseGroup(t, "NoCycle", tests)
-}
-
-func TestHazardCycleChecksCycle(t *testing.T) {
-	tests := []hazardCycleChecksTestCase{
 		{
-			name: "2-cycle in single chain with first log",
+			// 0->1->2->0
+			// 0->2->0
+			name: "3-cycle in single chain, 2-cycle in single chain with first log",
```

Comment:
```suggestion
			// 0->2->0   - both the executing messages
			// 0->2->1->0  - first executing message combined with regular log-ordering dependencies
			name: "3-cycle in single chain, 2-cycle in single chain with first log",
```

Log 0 here never depends directly on log 1. It depends on log 2. I think this go comment needs an update.